src/adaXT/parallel.py

```diff
@@ -0,0 +1,102 @@
+from functools import partial
+import multiprocessing
+from multiprocessing import cpu_count
+from itertools import starmap
+from typing import Any, Callable, Iterable
+
+from multiprocessing import RawArray
+import numpy as np
+import ctypes
+
+
+def shared_numpy_array(array):
+    if array.ndim == 2:
+        row, col = array.shape
+        shared_array = RawArray(ctypes.c_double, (row * col))
+        shared_array_np = np.ndarray(
+            shape=(row, col), dtype=np.double, buffer=shared_array
+        )
+    else:
```

Comment:
this else if for array.ndim == 1 correct. Maybe put a elif check here and raise an error?